devlib/utils/android.py

```diff
@@ -204,7 +205,12 @@ def push(self, source, dest, timeout=None):
         if timeout is None:
             timeout = self.timeout
         command = "push '{}' '{}'".format(source, dest)
-        return adb_command(self.device, command, timeout=timeout)
+        try:
+            return adb_command(self.device, command, timeout=timeout)
+        except CalledProcessError:
```

Comment:
Actually, `os.path.exists` is probably really fast, maybe we can just check before trying the ADB command - what do you think?